### PR TITLE
[Feature] Support function strcmp

### LINKED ISSUE #13246: [function]strcmp need to be suppported
[STRCMP()](https://dev.mysql.com/doc/refman/5.6/en/string-comparison-functions.html#function_strcmp) returns 0 if the strings are the same, -1 if the first argument is smaller than the second according to the current sort order, and 1 otherwise.
```
mysql> SELECT STRCMP('text', 'text2');
        -> -1
mysql> SELECT STRCMP('text2', 'text');
        -> 1
mysql> SELECT STRCMP('text', 'text');
        -> 0
```

### PR DESCRIPTION
**Pull request by tomscut**
Signed-off-by: Tao Li <tomscut@apache.org>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13246

## Problem Summary(Required) ：
Support function strcmp. This function compares two strings. Returns 0 if lhs and rhs compare equal. Return -1 if lhs appears before rhs in lexicographical order. Returns 1 if lhs appears after rhs in lexicographical order. 

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [x] I have added documentation for my new feature or new function

### CODE REVIEW

File: be/src/exprs/vectorized/string_functions.cpp [hunk 1]
```diff
@@ -2147,6 +2147,18 @@ ColumnPtr StringFunctions::get_char(FunctionContext* context, const Columns& col
     return VectorizedStringStrictUnaryFunction<get_charImpl>::evaluate<TYPE_INT, TYPE_CHAR>(columns[0]);
 }
 
+// strcmp
+DEFINE_BINARY_FUNCTION_WITH_IMPL(strcmpImpl, lhs, rhs) {
+    if (lhs == rhs) {
+        return 0;
+    }
+    return lhs > rhs ? 1 : -1;
```
Comment (imay) [inline be/src/exprs/vectorized/string_functions.cpp:2155]: if lhs is not equal with rhs, this function compares these two strings twice, which is not efficient.
There is already compare function in Slice, which you can use now.

Comment (tomscut) [reply be/src/exprs/vectorized/string_functions.cpp]: Thanks @imay for your comments. I updated it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Comment by imay** [comment]
run starrocks_admit_test

**Pull request merged**